lib/content-services/src/lib/common/services/saved-searches.service.ts

```diff
@@ -0,0 +1,138 @@
+/*!
+ * @license
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NodesApi, NodeEntry, SearchApi, SEARCH_LANGUAGE, ResultSetPaging } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { Observable, of, from, ReplaySubject, throwError } from 'rxjs';
+import { catchError, concatMap, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { AlfrescoApiService } from '../../services/alfresco-api.service';
+import { SavedSearch } from '../interfaces/saved-search.interface';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SavedSearchesService {
+    private _searchApi: SearchApi;
+    get searchApi(): SearchApi {
+        this._searchApi = this._searchApi ?? new SearchApi(this.apiService.getInstance());
+        return this._searchApi;
+    }
+
+    private _nodesApi: NodesApi;
+    get nodesApi(): NodesApi {
+        this._nodesApi = this._nodesApi ?? new NodesApi(this.apiService.getInstance());
+        return this._nodesApi;
+    }
+
+    private savedSearchFileNodeId: string;
+    private createFileAttempt = false;
+
+    readonly SAVED_SEARCHES_NODE_ID = 'saved-searches-node-id';
+
+    savedSearches$ = new ReplaySubject<SavedSearch[]>(1);
+
+    constructor(private apiService: AlfrescoApiService) {
+        this.getSavedSearches()
+            .pipe(take(1))
+            .subscribe((searches) => this.savedSearches$.next(searches));
+    }
+
+    getSavedSearches(): Observable<SavedSearch[]> {
+        return this.getSavedSearchesNodeId().pipe(
+            concatMap((nodeId) => {
+                this.savedSearchFileNodeId = nodeId;
+                return from(
+                    this.nodesApi.getNodeContent(nodeId).then((content) => {
+                        return this.mapFileContentToSavedSearches(content);
+                    })
+                ).pipe(
+                    catchError((error) => {
+                        if (!this.createFileAttempt) {
+                            this.createFileAttempt = true;
+                            localStorage.removeItem(this.SAVED_SEARCHES_NODE_ID);
+                            return this.getSavedSearches();
+                        }
+                        return throwError(error);
```

Comment:
Hmm, looks like now throwError is deprecated. Can we find alternative?
<img width="897" alt="image" src="https://github.com/user-attachments/assets/5c688d98-d826-4876-a5af-a429ac74f538">

Reply:
throwError is fine, it just need factory instead of value in constructor. thanks for the catch